src/reactviews/pages/QueryResult/queryResultPane.tsx

```diff
@@ -372,6 +404,17 @@ export const QueryResultPane = () => {
                         </Table>
                     </div>
                 )}
+                {metadata.tabStates!.resultPaneTab ===
+                    qr.QueryResultPaneTabs.ExecutionPlan &&
+                    Object.keys(metadata.resultSetSummaries).length > 0 && (
+                        <div
+                            id={"executionPlanResultsTab"}
+                            className={classes.queryResultContainer}
+                            style={{ height: "100%", minHeight: "300px" }}
+                        >
+                            <ExecutionPlanPage context={state} />
```

Comment:
I don't think passing context in this way is the best design approach. It looks like you've special-cased the context variable in all the ExecutionPlan children components:
```
var context = context ?? useContext(ExecutionPlanProvider)
```
The purpose of using context is to avoid prop drilling (i.e., passing props down through multiple child components). A more effective way to handle this would be:
```
<ExecutionPlanContextProvider
   props={queryResultState.executionPlan}
>
   <ExecutionPlanPage />
</ExecutionPlanContextProvider>
```
By properly initializing all the props in the provider, you wouldn't need to modify the existing ExecutionPlan components.